### PR TITLE
Fixes SI-6337 by disallowing nested value classes.

### PR DESCRIPTION
**Pull request by odersky**
It seems for the moment too hard to allow this, and the functionality to have value classes wrap other value classes does not seem essential. Review by @paulp, @harrah

### CODE REVIEW

File: src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala [hunk 1]
```diff
@@ -135,7 +135,8 @@ abstract class ExtensionMethods extends Transform with TypingTransformers {
       tree match {
         case Template(_, _, _) =>
           if (currentOwner.isDerivedValueClass) {
-            checkNonCyclic(currentOwner.pos, Set(), currentOwner)
+          /* This is currently redundant since value classes may not
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala:138]: may not what?

Comment (gkossakowski) [inline src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala:138]: @xeno-by: I don't understand your comment.

**Comment by harrah** [comment]
Seems to do what it says it does. LGTM

I am assuming that https://issues.scala-lang.org/browse/SI-6337 stands on its own and needs a fix like this, while issues like https://issues.scala-lang.org/browse/SI-6385 do not share the same fundamental cause and so there is no common fix that would avoid the need for this restriction.

**Comment by odersky** [comment]
Yes, SI-6337 does seem to stand on its own. There's hope we can allow nested value classes at some point in the fture, but right now it's most prudent to disallow.

**Comment by paulp** [comment]
LGTM.

**Pull request merged**